apiserver/logsink/logsink_test.go

```diff
@@ -60,7 +60,9 @@ func (s *logsinkSuite) SetUpTest(c *gc.C) {
 	s.abort = make(chan struct{})
 	s.written = make(chan params.LogRecord, 1)
 	s.stub.ResetCalls()
+	s.stackMu.Lock()
```

Comment:
If this is an issue, then there's still an issue, because the only time I can see this happening is if a previous test is still running while SetUpTest has been called, which means there's something not torn down correctly. That seems potentially problematic to me.

Given that this is only for a log anyway, why not just drop the log (and lastStack too) or, if the logged information is deemed important enough, just change recordStack to log the stack directly rather than setting an instance variable?